Example/src/TagInputExample.js

```diff
@@ -16,18 +19,40 @@ export default class TagInputExample extends Component {
     });
   };
 
+  onChangeHorizontalTags = (horizontalTags) => {
+    this.setState({
+      horizontalTags,
+    });
+  };
+
   labelExtractor = (tag) => tag;
 
+  onParsePendingInput = () => {
+    this.refs.horizontalTagInput.parseTags();
```

Comment:
I'm not a fan of having `TagInput` hold on to a piece of state - in this case, the `text` state - that its parent needs to manipulate.

In situations where multiple React components need to manipulate state, that state should be ["lifted up to their closest common ancestor"](https://reactjs.org/docs/lifting-state-up.html). In other words - if the parent component and child component both need to maintain a piece of state, then the parent should be holding on to that state.

In general, refs are a hack around this, and should be used sparingly and only when necessary.

More discussion in my higher-level review comment.

Reply:
👍